### PR TITLE
[LETS-121] - Inconsistency between packing and unpacking of VPID

### PR DESCRIPTION
**Pull request by danilailie**
http://jira.cubrid.org/browse/LETS-121

* Fixed inconsistency between packing and unpacking of VPID

### CODE REVIEW

File: src/server/page_server.cpp [hunk 1]
```diff
@@ -110,10 +110,21 @@ void page_server::receive_data_page_fetch (cubpacking::unpacker &upk)
...
+
+  std::memcpy (&vpid.volid, message.c_str () + bytes_read, sizeof (vpid.volid));
+  bytes_read += sizeof (vpid.volid);
 
   LOG_LSA target_repl_lsa;
-  std::memcpy (&target_repl_lsa, message.c_str () + sizeof (vpid), sizeof (target_repl_lsa));
+  std::memcpy (&target_repl_lsa, message.c_str () + bytes_read, sizeof (target_repl_lsa));
+  bytes_read += sizeof (target_repl_lsa);
+
+  _er_log_debug (ARG_FILE_LINE,
```
Comment (cristiarg) [inline src/server/page_server.cpp]: The debug lines should be removed. They were used just for debugging. And `bytes_read` goes down too.

File: src/storage/page_buffer.c [hunk 2]
```diff
@@ -7868,6 +7868,11 @@ pgbuf_request_data_page_from_page_server (const VPID * vpid)
       std::memcpy (buffer + bytes_copied, &nxio_lsa, sizeof (nxio_lsa));
       bytes_copied += sizeof (nxio_lsa);
 
+      _er_log_debug (ARG_FILE_LINE,
```
Comment (cristiarg) [inline src/storage/page_buffer.c]:
```
      constexpr size_t INT32_SIZE = 4;
      constexpr size_t SHORT_SIZE = 2;
```
This is not portable code. These sizes are not guaranteed. Better:
```
      constexpr size_t PAGEID_SIZE = sizeof (VPID::pageid);
      constexpr size_t VOLID_SIZE = sizeof (VPID::volid);
```

**Review by cristiarg — COMMENTED** [review]
Maybe it is better to add a utility class that inherits from `cubpacking::packable_object` and does the pack/unpack. Then this class could be unit tested to check the forth-back-forth validity of pack/unpack.

File: src/server/page_server.cpp [hunk 3]
```diff
@@ -110,10 +110,16 @@ void page_server::receive_data_page_fetch (cubpacking::unpacker &upk)
   upk.unpack_string (message);
 
   VPID vpid;
-  std::memcpy (&vpid, message.c_str (), sizeof (vpid));
+  int bytes_read = 0;
```
Comment (ribram) [inline src/server/page_server.cpp]:
```suggestion
  size_t bytes_read = 0;
```

**Review by ribram — APPROVED** [review]

**Pull request merged**